src/Badge/Badge.component.tsx

```diff
@@ -2,45 +2,148 @@
+    standalone?: boolean;
+    textColor?: string;
+}
+
+interface BadgeStylesGroup {
+    [key: string]: FlattenSimpleInterpolation;
 }
 
+const BadgeSizeStyles: BadgeStylesGroup = {
+    dot: css`
+        height: 0.75rem;
+        width: 0.75rem;
+    `,
+    small: css`
+        font-size: 0.625rem;
+        height: 1.25rem;
+        min-width: 1.25rem;
+        padding: 0 0.25rem;
+    `,
+    large: css`
+        font-size: 0.75rem;
+        height: 1.75rem;
+        min-width: 1.75rem;
+        padding: 0 0.5rem;
+    `,
+};
+
 const StyledBadgeContainer = styled.div`
     position: relative;
     font-family: ${fonts.fontFamily};
     display: inline-block;
 `;
 
-const StyledBadge = styled.div`
-    position: absolute;
-    top: -0.5rem;
-    right: -0.25rem;
-    background-color: ${colors.flashPink.base};
-    color: ${colors.white.base};
+const StyledBadge = styled.div<BadgeProps>`
+    align-items: center;
+    background-color: ${({ backgroundColor }: BadgeProps) => backgroundColor};
+    border: 0.125rem solid ${({
+        borderColor,
+        borderColorHover,
+        isParentHovered,
+    }: BadgeProps) => (isParentHovered ? borderColorHover : borderColor)};
+    box-sizing: border-box;
+    color: ${({ textColor }: BadgeProps) => textColor};
+    display: flex;
     font-weight: 700;
-    font-size: 0.75rem;
-    padding: 1px ${sizes.padding.xs};
-    border-radius: ${variables.borderRadius};
+    justify-content: center;
+    text-align: center;
+
+    /* Border Radius */
+    ${({ count = 0, size = 'dot' }: BadgeProps) =>
+        size === 'small' && count < 10
+            ? `border-radius: 0.4rem;`
+            : `border-radius: 1rem;`}
+
+    /* Non-standalone styles */
+    ${({ offsetBottom, offsetLeft, standalone = false }: BadgeProps) =>
+        standalone
+            ? null
+            : `
+      position: absolute;
+      bottom: ${offsetBottom}rem;
+      left: ${offsetLeft}rem;
+    `}
+
+    /* Sizes */
+    ${({ size = 'large' }: BadgeProps) => BadgeSizeStyles[size]}
+
+    &:hover {
+        border: 0.125rem solid ${({ borderColorHover }: BadgeProps) =>
+            borderColorHover};
+    }
 `;
 
 export const Badge = ({
-    className,
+    backgroundColor = colors.flashPink.base,
+    borderColor,
+    borderColorHover,
     children,
-    count,
+    className,
+    count = 0,
+    isParentHovered,
+    overflowCount = 9,
+    offsetBottom,
+    offsetLeft,
+    size = 'dot',
+    standalone = false,
+    textColor = colors.white.base,
     ...props
-}: BadgeProps): JSX.Element => (
-    <StyledBadgeContainer
-        className={classNames('anchor-badge', className)}
-        {...props}
-    >
-        {children}
-        <StyledBadge>{count || '•'}</StyledBadge>
-    </StyledBadgeContainer>
-);
+}: BadgeProps): JSX.Element => {
+    const getDisplayCount = (
+        badgeCount: number,
+        badgeOverflowCount: number
+    ) => {
+        if (badgeCount === 0) {
+            return '';
```

Comment:
This was a desired behavior: '•'

Reply:
Let me sync with Jen on this. I think she said she didn't want that, though let's make sure we have all the cases covered.

Reply:
@sharkparty I talked with Jen and she said she does not want a `•`.  

Our did bring up an uncovered case though. She wants the default to be that the Badge does not show if the `count` is `0`, but she also wants the option to render a Badge with a value of `0`, so I'm planning to add a `showZero` prop that defaults to `false` like Ant does.

Reply:
Engineering wants the ability to use that • as a value. So long as that is possible, I'm cool with this implementation.

Reply:
Sure, we can make that possible. Can you tell me more about when we would use `•` instead of a number? I'm wondering if I need to connect it to a certain state of the component or just display it correctly if it is passed in via the `count` prop.

Reply:
It was a request made by Andy during the UI Guild that I added. I think it's how we show that badge on RMN core if we don't have a count available but do know that there is a notification. I really didn't push back on it bc it was pretty trivial to add.

Reply:
Hey Kam, I just chatted with Andy and he said that using the dot size of the badge would work for the use case you mentioned. Are you ok with that solution?

Reply:
Absolutely. He’s the stakeholder, so if he opts into something else then that’s great!